lib/FrontendTool/FrontendTool.cpp

```diff
@@ -2811,6 +2831,22 @@ int swift::performFrontend(ArrayRef<const char *> Args,
     return finishDiagProcessing(1, /*verifierEnabled*/ false);
   }
 
+  llvm::StringMap<std::vector<std::string>> FileSpecificDiagnostics;
+  std::unique_ptr<DiagnosticConsumer> FileSpecificAccumulatingConsumer;
+  if (Invocation.getFrontendOptions().FrontendParseableOutput) {
+    // We need a diagnostic consumer that will, per-file, collect all
+    // diagnostics to be reported in parseable-output
+    FileSpecificAccumulatingConsumer = createAccumulatingDiagnosticConsumer(
+        Invocation.getFrontendOptions().InputsAndOutputs,
+        FileSpecificDiagnostics);
+    Instance->addDiagnosticConsumer(FileSpecificAccumulatingConsumer.get());
+
+    // If we got this far, we need to suppress the output of the
+    // PrintingDiagnosticConsumer to ensure that only the parseable-output
+    // is emitted
+    PDC.setSuppressOutput(true);
```

Comment:
Just checking: We're disabling the PrintingDiagnosticConsumer instead of never adding it in the first place because we need the PDC while we're parsing arguments, including the argument that sets this option?

Reply:
Exactly. In fact, it seems that this is why 'setSuppressOutput' exists in the first place.